test/db/SQLJSPackageDB.test.ts

```diff
@@ -0,0 +1,589 @@
+          resourceType: 'StructureDefinition',
+          id: 'my-patient-profile'
+        })
+      );
+    });
+
+    it('should save a resource with additional information', () => {
+      packageDb.saveResourceInfo({
+        resourceType: 'ValueSet',
+        id: 'my-value-set',
+        url: 'http://example.org/ValueSets/my-value-set',
+        name: 'MyValueSet',
+        version: '3.1.3',
+        packageName: 'RegularPackage',
+        packageVersion: '3.2.2',
+        resourcePath: '/var/data/.fhir/RegularPackage-3.2.2/ValueSets/my-value-set.json'
+      });
+      expect(dbRunSpy).toHaveBeenCalledTimes(1);
+      const resource = packageDb.findResourceInfo('my-value-set');
+      expect(resource).toEqual(
+        expect.objectContaining({
+          resourceType: 'ValueSet',
+          id: 'my-value-set',
+          url: 'http://example.org/ValueSets/my-value-set',
+          name: 'MyValueSet',
+          version: '3.1.3',
+          packageName: 'RegularPackage',
+          packageVersion: '3.2.2',
+          resourcePath: '/var/data/.fhir/RegularPackage-3.2.2/ValueSets/my-value-set.json'
+        })
+      );
+    });
+
+    it('should save a resource with additional structure definition information', () => {
+      packageDb.saveResourceInfo({
+        resourceType: 'StructureDefinition',
+        id: 'my-patient-profile',
+        name: 'MyPatientProfile',
+        sdKind: 'resource',
+        sdDerivation: 'constraint',
+        sdType: 'Patient',
+        sdBaseDefinition: 'http://hl7.org/fhir/StructureDefinition/Patient',
+        sdAbstract: false,
+        sdImposeProfiles: [
+          'http://example.org/SomeOtherPatient',
+          'http://example.org/YetAnotherPatient'
+        ],
+        // a profile wouldn't normally have characteristics,
+        // but they are used here for test purposes
+        sdCharacteristics: ['can-be-target', 'do-translations'],
+        sdFlavor: 'Profile',
+        packageName: 'RegularPackage',
+        packageVersion: '3.2.2'
+      });
+      expect(dbRunSpy).toHaveBeenCalledTimes(1);
+      const resource = packageDb.findResourceInfo('my-patient-profile');
+      expect(resource).toEqual(
+        expect.objectContaining({
+          resourceType: 'StructureDefinition',
+          id: 'my-patient-profile',
+          name: 'MyPatientProfile',
+          sdKind: 'resource',
+          sdDerivation: 'constraint',
+          sdType: 'Patient',
+          sdBaseDefinition: 'http://hl7.org/fhir/StructureDefinition/Patient',
+          sdAbstract: false,
+          sdImposeProfiles: [
+            'http://example.org/SomeOtherPatient',
+            'http://example.org/YetAnotherPatient'
+          ],
+          sdCharacteristics: ['can-be-target', 'do-translations'],
+          sdFlavor: 'Profile',
+          packageName: 'RegularPackage',
+          packageVersion: '3.2.2'
+        })
+      );
+      // ?? do we want to turn it back into a boolean, or is leaving it as a 0 good enough
```

Comment:
What is this comment referencing? Is it perhaps related to how `sdAbstract` worked before you fixed it in the implementation?

Reply:
Sure is! I've removed it.